google/api_core/retry.py

```diff
@@ -54,13 +54,14 @@ def check_if_exists():
 
 """
 
-from __future__ import unicode_literals
+from __future__ import annotations
```

Comment:
this library has `python_requires >= 3.7` so (1) `unicode_literals` does nothing and (2) this is the first version which supports `__future__.annotations`

it might be a good idea to run https://github.com/asottile/pyupgrade on this codebase to remove the remaining python 2isms

Reply:
Thanks! I've captured this TODO in https://github.com/googleapis/python-api-core/issues/524